File: json/php/update/json_update.php
#! /usr/bin/php
<?php
// ------------------------------------------------------------------
//	update/json_update.php
//
//					Sep/15/2016
//
// ------------------------------------------------------------------
$path = "/var/www/data_base/common/php_common";

set_include_path (get_include_path() . PATH_SEPARATOR . $path);
//
include "file_io.php";
include "text_manipulate.php";
// ------------------------------------------------------------------
fputs (STDERR,"*** 開始 ***\n");

$json_filename = $argv[1];
$id_in = $argv[2];
$population_in = $argv[3];

echo	$json_filename . "\n";
echo	$id_in . "\t";
echo	$population_in . "\n";

$json_string = file_get_contents ($json_filename);
$dict_aa = json_decode ($json_string,true);

// var_dump ($dict_aa);

$dict_bb = dict_update_proc ($dict_aa,$id_in,$population_in);

$js_out = json_encode ($dict_bb);

file_write_proc ($js_out,$json_filename);

fputs (STDERR,"*** 終了 ***\n");

// ------------------------------------------------------------------
?>

File: json/dart/update/json_update.dart
/*
	json_update.dart

					Jan/09/2021
*/
// -------------------------------------------------------------
import 'dart:io';
import 'dart:convert';

import	'/var/www/data_base/common/dart_common/text_manipulate.dart';
// -------------------------------------------------------------
main (List<String> arguments) {
	stderr.writeln ("*** 開始 ***");

	String file_name = '${arguments[0]}';
	String key = '${arguments[1]}';
	int population = int.parse ('${arguments[2]}');

	print (key + '\t' + population.toString ());

	File file = new File (file_name);
	Future future = file.readAsString ();

	future.then((content)
		{
		var dict_aa = jsonDecode(content);

		if (dict_aa.containsKey (key))
			{
			dict_aa = dict_update_proc (dict_aa,key,population);

			var str_out = jsonEncode(dict_aa);

			str_to_file_proc (str_out,file_name);
			}

		stderr.writeln ("*** 終了 ***");
		}); 
}

// -------------------------------------------------------------
